huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go

```diff
@@ -234,50 +256,48 @@ func ResourceAlarmRule() *schema.Resource {
 	}
 }
 
-func getMetricOpts(d *schema.ResourceData) (alarmrule.MetricOpts, error) {
-	mos, ok := d.Get("metric").([]interface{})
-	if !ok {
-		return alarmrule.MetricOpts{}, fmtp.Errorf("Error converting opt of metric:%v", d.Get("metric"))
-	}
-	mo := mos[0].(map[string]interface{})
-
-	mod := mo["dimensions"].([]interface{})
-	dopts := make([]alarmrule.DimensionOpts, len(mod))
-	for i, v := range mod {
-		v1 := v.(map[string]interface{})
-		dopts[i] = alarmrule.DimensionOpts{
-			Name:  v1["name"].(string),
-			Value: v1["value"].(string),
+func getMetricOpts(d *schema.ResourceData) alarmrule.MetricOpts {
+	metricRaw := d.Get("metric").([]interface{})
+	metric := metricRaw[0].(map[string]interface{})
```

Comment:
suggest to check the length of metricRaw even if "metric" is required

Reply:
done